Remove highlight.js typings

WilsonGramer opened the pull request:
`highlight.js` provides its own types: https://github.com/highlightjs/highlight.js/blob/master/types/index.d.ts

Because the `markdown-it` and `remarkable` type definitions depend on these typings, this pull request cannot be merged until https://github.com/microsoft/DefinitelyTyped-tools/pull/151 has been merged.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If removing a declaration:
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

ffflorian commented:
Okay but do the official types also support importing `highlightjs` (which leads to the same package)? If not, please leave the types for `highlightjs` where they are and edit them instead to import `highlight.js`. Thanks!

WilsonGramer commented:
@ffflorian Sure, I just added them back. Thanks for pointing that out!

peterblazejewicz commented on types/remarkable/package.json:
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "highlight.js": "^10.1.0"
```
I had a similar problem, this cannot be just done by adding an updated dependency to 
DT for older version of package,
the current DT version is supporting v9, not v10:
https://github.com/jonschlinkert/remarkable/blob/v1.7.4/package.json#L74

peterblazejewicz commented on types/markdown-it/package.json:
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "highlight.js": "^10.1.0"
```
_ditto_
here DT is at v10.0, which is based on removed version:
https://github.com/markdown-it/markdown-it/blob/10.0.0/package.json#L43
it just happens there is no compiler errors.

nikeee commented:
There may still be issues with the types that ship with hljs itself.

https://github.com/highlightjs/highlight.js/issues/2603

Don't know the status of that.

WilsonGramer commented:
I just updated the dependencies using old versions of `highlight.js` and I'm not getting type errors anymore. Since `remarkable` uses the typings for `highlight.js@^9.7.0`, I added `@typings/highlight.js` to the list of dependencies under `types/remarkable/package.json`. Now I get this error:

```
Error: In package.json: Dependency @types/highlight.js not in the allowed dependencies list.
Don't use a 'package.json' for @types dependencies unless this package relies on
an old version of types that have since been moved to the source repo.
For example, if package *P* used to have types on Definitely Typed at @types/P,
but now has its own types, a dependent package *D* will need to use package.json
to refer to @types/P if it relies on old versions of P's types.
In this case, please make a pull request to microsoft/DefinitelyTyped-tools adding @types/P to `packages/definitions-parser/allowedPackageJsonDependencies.txt`.
```

So it looks like I need to submit another pull request to add the old `@types/highlight.js` (since this branch removes it from DefinitelyTyped). Is this the right thing to do, or is there a different way I should be referencing the old typings?

peterblazejewicz reviewed (approved):
LGTM!
@WilsonGramer thx!
:shipit:

joshgoebel commented:
Is this strange `namespace` + `export function` and `export var` the only way to include both type exports AND a default export in the same file?  I see how it's kind of mixing everything into one huge blob (which also seems a bit strange to me).

https://github.com/highlightjs/highlight.js/issues/2945

Is there some better way for us to do this?  I find the whole ambient module thing very confusing.  If anyone can assist we'd be grateful.
